server/errorstore/errors.go

```diff
@@ -134,74 +134,17 @@ func sha256b64(s string) string {
 }
 
 func hashError(err error) string {
-	// Ok so the hashing process is as follows:
```

Comment:
decided to remove this comment because `c` and `d` are no longer the case if we get rid of eris and `a` and `b` seemed to be at odds. Specifically:

> a) [ ... ] so that if by mistake the same error is sent to
> Handler.Handle in multiple places in the code, after being wrapped
> differently any number of times, it still hashes to the same value
> (because the root is the same)

> b) [ ... ] we also want to hash all locations in the stack
> trace, so that the same error type and message [...]
> caused at two different places in the code are not
> considered the same error.

am I misunderstanding this?

Reply:
Hm, that's indeed contradictory... Should we incline towards b?

Reply:
For reference we're having a very similar conversation in the [review comments](https://github.com/fleetdm/fleet/pull/5733#discussion_r874783786). I see your point, but you lose some information, I'm thinking of this example (which might not be relevant for us, if so please lmk)

```go
func handlerWithWrongParams() {
  if _, err := dataStoreMethod(wrongParam); err != nil {
    return ctxerr.Wrap(err, "from handlerWithWrongParams")
  }
}

func anotherHandler() {
  if _, err := dataStoreMethod(param); err != nil {
    return ctxerr.Wrap(err, "from anotherHandler")
  }
}

func dataStoreMethod(param string) ([]any, error) {
  // ...
  return empty, errors.New("error retrieving data")
}
```

I kind of want to have both errors hashed (and thus stored) differently so I can see where the problem is. Another way to think about my point is: instead of wrapping, think you're actually modifying the message and the stack trace every time you call `Wrap` and similar methods.

This is assuming we're not calling `handler.Store` on each of the wraps, and only in the final layer

Reply:
That's a great example (assuming in `dataStoreMethod` you actually want to call `ctxerr.New`, so that it does capture the stack trace at that point, right?). You would have two different errors stored in this case, as the stack trace is not the same, one is `anoterHandler->dataStoreMethod`, and the other is `handlerWithWrongParams->dataStoreMethod`. This is fine and not the same error based on code path, that's not what I'm arguing that should become the same error.

Reply:
gotcha, I see what you say! thanks for talking this through. We just need to ensure special handling of errors that are not `FleetError`